File: src/main/java/my/study/repository/design/pattern/bridge/java/coffee/SmallCoffee.java
package my.study.repository.design.pattern.bridge.java.coffee;

import my.study.repository.design.pattern.bridge.java.additive.BaseAdditive;

/**
 * @author djh on  2018/9/25 14:16
 * @E-Mail 1544579459@qq.com
 */
public class SmallCoffee extends BaseCoffee {

    public SmallCoffee(BaseAdditive baseAdditive) {
        super(baseAdditive);
    }

    @Override
    public void showCoffeeInfo() {
        System.out.println("中杯加 " + mBaseAdditive.toString() + " 咖啡");
    }
}

File: src/main/java/my/study/repository/design/pattern/bridge/kotlin/coffee/SmallCoffee.kt
package my.study.repository.design.pattern.bridge.kotlin.coffee

import my.study.repository.design.pattern.bridge.kotlin.additive.BaseAdditive


/**
 * @author djh on  2018/10/4 17:48
 * @E-Mail 1544579459@qq.com
 */
class SmallCoffee(baseAdditive: BaseAdditive) : BaseCoffee(baseAdditive) {

    override fun showCoffeeInfo() {
        println("小杯 $baseAdditive 咖啡")
    }

}